test/labeling/model/test_label_model.py

```diff
@@ -371,9 +381,39 @@ def test_label_model(self) -> None:
         np.testing.assert_array_almost_equal(P, P_lm, decimal=2)
 
         # Test predicted labels
-        Y_lm = label_model.predict_proba(L).argmax(axis=1)
-        err = np.where(Y != Y_lm, 1, 0).sum() / self.n
-        self.assertLess(err, 0.1)
+        score = label_model.score(L, Y)
+        self.assertGreaterEqual(score["accuracy"], 0.9)
+
+    def test_label_model_sparse(self) -> None:
+        """Test the LabelModel's estimate of P and Y on a sparse synthetic dataset.
+
+        This tests the common setting where LFs abstain most of the time, which can
+        cause issues for example if parameter clamping set too high (e.g. see Issue
+        #1422).
+        """
+        np.random.seed(123)
+        P, Y, L = generate_simple_label_matrix(
+            self.n, self.m, self.cardinality, abstain_multiplier=1000.0
+        )
+
+        # Train LabelModel
+        label_model = LabelModel(cardinality=self.cardinality, verbose=False)
+        label_model.fit(L, n_epochs=1000, lr=0.01, seed=123)
+
+        # Test estimated LF conditional probabilities
+        P_lm = label_model._get_conditional_probs().reshape(
+            (self.m, self.cardinality + 1, -1)
+        )
+        np.testing.assert_array_almost_equal(P, P_lm, decimal=2)
+
+        # Test predicted labels *only on non-abstained data points*
+        Y_pred = label_model.predict(L, tie_break_policy="abstain")
+        idx, = np.where(Y_pred != -1)
+        acc = np.where(Y_pred[idx] == Y[idx], 1, 0).sum() / len(idx)
+        self.assertGreaterEqual(acc, 0.65)
```

Comment:
this feels like a pretty low bar. what is the accuracy without the clamping patch?

Reply:
It's a hard problem when the LFs are this sparse; no way around that.  Without the clamping patch, the accuracy as measured here- not counting abstains- is roughly the same but coverage is far lower because of the error where LFs are counted as having null signal

Reply:
Ok cool so `test_label_model_sparse` fails without this patch?

Reply:
Yes